src/contracts/core/AVSDirectory.sol

```diff
@@ -417,9 +419,9 @@ contract AVSDirectory is
                 "AVSDirectory._deregisterOperatorFromOperatorSet: operator not registered for operator set"
             );
 
-            --operatorSetMemberCount[avs][operatorSetIds[i]];
-
-            _operatorSetsMemberOf[operator].remove(_encodeOperatorSet(operatorSet));
+            bytes32 encodedOperatorSet = _encodeOperatorSet(operatorSet);
+            _operatorSetsMemberOf[operator].remove(encodedOperatorSet);
```

Comment:
hm. instead of the is member check you could acually just

```solidity
require(_operatorSetsMemberOf[operator].remove(encodedOperatorSet))
```

i think

Reply:
Yup, good suggestion. Updated reg and dereg methods